packages/firestore/src/util/misc.ts

```diff
@@ -24,14 +24,14 @@ export type EventHandler<E> = (value: E) => void;
  * unknown. Unlike "any" this doesn't lose all type-safety, since the consuming
  * code must still cast to a particular type before using it.
  */
-export type AnyJs = null | undefined | boolean | number | string | object;
+export type AnyJs = unknown;
 
 /**
  * `Unknown` is a stand-in for Typescript 3's `unknown` type. It is similar to
  * `any` but forces code to check types before performing operations on a value
  * of type `Unknown`. See: https://blogs.msdn.microsoft.com/typescript/2018/07/30/announcing-typescript-3-0/#the-unknown-type
  */
-export type Unknown = null | undefined | {} | void;
+export type Unknown = unknown;
```

Comment:
@mikelehen Can Unknown and AnyJs be replaced by 'unknown'? If it sounds right to you, I will make a another PR to remove AnyJs and Unknown and replace all occurrences with `unknown`

Reply:
Yes, that sounds right to me.  But let's do the massive search/replace in a separate PR (I can take care of it if you like).  For now, what you've done here looks good.  You could add a TODO to delete AnyJs and Unknown if you like.

Reply:
Sure. I will create a separate PR for this.